testif.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+# This script is invoked by Travis testing and inspects
+# the changeset that's being built.
+#
+# Usage:
+#
+#    testif.sh <label> && something_else
+#
+# The intention is to be able to ONLY test certain aspects
+# when special conditions are satisfied.
+#
+# Notes:
+#    JS tests are expected to be very fast
+#    We shouldn't skip tests when it affects Coverage negatively
+#    We can possibly do tests based on which branch is pushed to
+#
+# https://github.com/learningequality/kolibri/issues/27
+#
+# What changes are analyzed?
+#
+# For this script, we should mainly use Travis' env
+# $TRAVIS_COMMIT_RANGE. The reason is simple: You cannot tell what
+# changes have been implemented since last test, thus if your
+# condtion should fire or not.
+#
+# Definition:
+# TRAVIS_COMMIT_RANGE: The range of commits that were included in
+# the push or pull request.
+#
+# https://docs.travis-ci.com/user/environment-variables/
+
+
+set -e
+
+LABEL="$1"
+
+branches_to_always_test=( "master" "releases/*" )
+
+# Set by Travis CI
+commit="$TRAVIS_COMMIT"
+current_branch="$TRAVIS_BRANCH"
+
+git_changeset=""
+
+if [ "$TRAVIS_PULL_REQUEST" == "false" ]
+then
+    # The changeset is $TRAVIS_COMMIT_RANGE
+    # echo "Using TRAVIS_COMMIT_RANGE as changeset"
+    git_changeset=`git show --name-only --no-notes --oneline $TRAVIS_COMMIT_RANGE`
+else
```

Comment:
After this `else`, there was a `:` which I didn't spot since it's not strictly syntactically wrong. It took sooooo many hours to figure out what was wrong :(